src/gl-matrix/vec3.js

```diff
@@ -522,29 +522,40 @@ export function transformMat3(out, a, m) {
 
 /**
  * Transforms the vec3 with a quat
+ * Can also be used for dual quaternions. (Multiply it with the real part)
  *
  * @param {vec3} out the receiving vector
  * @param {vec3} a the vector to transform
  * @param {quat} q quaternion to transform with
  * @returns {vec3} out
  */
 export function transformQuat(out, a, q) {
```

Comment:
I don't think we should rewrite this function to handle parameter overriding for both `quat` and `quat2` types. The `gl-matrix` library aims towards a 'functional programming' style of 'pure functions' which aims for static type style fixed function signatures.  I would propose a new `transformQuat2` method that explicitly handles the `quat2` behavior.

Reply:
:) I'm perfectly aware of it. 

This thing works because a dual quaternion is represented by a flat array with 2 quaternions. 
`[x0, y0, z0, w0,  x1, y1, z1, w1]` 
The first part is a perfectly normal, normalized quaternion which represents a rotation. The second part is a quaternion which represents a translation. 

And, the function only looks at the first 4 elements of whatever array gets passed to it. In our case, it's the dual quaternion's rotation. It ignores the translation part, which is perfectly fine, because [a vector doesn't really have a translation.](https://en.wikipedia.org/wiki/Euclidean_vector) (When you have a vector that represents a point, it's actually a vector from the origin to the point's coordinates.)

However, this comment might also cause some confusion. e.g.
> Why doesn't this translate my vector?
transformQuat(out, [0,0,0], [no rotation, a translation])

(The answer would be: Because a vector only has a "magnitude (or length) and direction" --Wikipedia)

So, I guess it can be removed? If a person actually wants to rotate his vectors with a dual quaternion's rotation part, he probably knows his stuff and doesn't need this comment?